scripts/establish_conda_env.sh

```diff
@@ -1,58 +1,294 @@
 #!/bin/bash
 
-#Similar to setup_raven_libs.sh, but installs libraries, not just opens environment
-SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# USES:
+# --load - just finds and sources the conda environment (default)
+# --install - find (create if not found) and update the environment, then load it
+# OTHER OPTIONS:
+# --optional - if updating, install optional libraries as well as base ones
 
-conda_install_or_create ()
+# ENVIRONMENT VARIABLES
+# location of conda definitions: CONDA_DEFS (defaults if not set based on OS)
+# name for raven libraries: RAVEN_LIBS_NAME (defaults to raven_libraries if not set)
+
+ECE_SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+RAVEN_UTILS=${ECE_SCRIPT_DIR}/TestHarness/testers/RavenUtils.py
+
+# fail if ANYTHING this script fails (mostly, there are exceptions)
+set -e
+
+function establish_OS ()
 {
-  if conda env list | grep -q raven_libraries;
-  then
-    echo ... RAVEN environment located, checking packages ...
-    `python $SCRIPT_DIR/TestHarness/testers/RavenUtils.py --conda-install`
-  else
-    echo ... No RAVEN environment located, creating it ...
-    try_using_raven_environment
-    `python $SCRIPT_DIR/TestHarness/testers/RavenUtils.py --conda-create`
-  fi
+	case $OSTYPE in
+		"linux")
+			OSOPTION="--linux"
+			;;
+		"linux-gnu")
+			OSOPTION="--linux"
+			;;
+		"darwin"*)
+			OSOPTION="--mac"
+			;;
+		"msys"*)
+			OSOPTION="--windows"
+			;;
+		"cygwin"*)
+			OSOPTION="--windows"
+			;;
+		*)
+			echo Unknown OS: $OSTYPE\; ignoring.
+			OSOPTION=""
+			;;
+	esac
 }
 
-try_using_raven_environment ()
+function find_conda_defs ()
 {
-  #First check home directory for a script, then /opt/raven_libs
-  if test -e $HOME/.raven/environments/raven_libs_profile;
+	if [ -z ${CONDA_DEFS} ];
+	then
+    # first check the RAVEN RC file for the key
+    CONDA_DEFS=`echo $(python ${ECE_SCRIPT_DIR}/update_install_data.py --read CONDA_DEFS)`
+    # if not setin RC, then will be empty string; next try defaults
+    if [[ ${#CONDA_DEFS} == 0 ]];
     then
-  source $HOME/.raven/environments/raven_libs_profile
+      # default location of conda definitions, windows is unsurprisingly an exception
+      if [[ "$OSOPTION" = "--windows" ]];
+      then
+        CONDA_DEFS="/c/ProgramData/Miniconda2/etc/profile.d/conda.sh";
+      else
+        CONDA_DEFS="$HOME/miniconda2/etc/profile.d/conda.sh";
+      fi
+    # if found in RC, just use that.
     else
-  if test -e /opt/raven_libs/environments/raven_libs_profile;
+      if [[ $ECE_VERBOSE == 0 ]];
+      then
+        echo ... found conda path in ravenrc: ${CONDA_DEFS}
+        echo ... \>\> If this is not the desirable path, rerun with argument --conda-defs \[path\] or remove the entry from raven/.ravenrc file.
+      fi
+    fi
+	fi
```

Comment:
not properly indented

Reply:
Indenting is not part of the bash syntax, and we don't have a standard for bash yet.